mbuild/formats/hoomdxml.py

```diff
@@ -149,16 +149,16 @@ def write_hoomdxml(structure, filename, forcefield, box, ref_distance=1.0, ref_m
 
         bonds = [[bond.atom1.idx, bond.atom2.idx] for bond in structure.bonds] 
         if bonds:
-            if len(set([bond.type for bond in structure.bonds])) == 1:
+            if len(structure.bond_types) == 0:
                 bond_types = np.zeros(len(bonds),dtype=int)
             else:
                 all_bond_types = dict(enumerate(set([(round(bond.type.k,3),
                                                       round(bond.type.req,3)) for bond in structure.bonds])))
-                all_bond_types = {y:x for x,y in all_bond_types.iteritems()}
+                all_bond_types = {y:x for x,y in all_bond_types.items()}
                 bond_types = [all_bond_types[(round(bond.type.k,3),
                                               round(bond.type.req,3))] for bond in structure.bonds]
                 xml_file.write('<bond_coeffs>\n')
-                for params,id in all_bond_types.iteritems():
+                for params,id in all_bond_types.items():
```

Comment:
One more minor suggestion: `id` is a python builtin. I would just change it to `idx` or something like that.

Happens a few times below as well